File: SIC_Backend/Data/Models/Section.cs
namespace SIC_Backend.Data.Models
{
    public class Section
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string UserId { get; set; }
        public int PlaceId { get; set; }

        public User? User { get; set; }
        public Place? Place { get; set; }
        public ICollection<Device> Devices { get; set; } = [];

        public static Section FromCreateModel(CreateSectionModel model)
        {
            return new Section
            {
                Name = model.Name,
                UserId = model.UserId,
                PlaceId = model.PlaceId
            };
        }
    }
}

File: sic_home/lib/models/section.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sic_home/models/device.dart';

part 'section.g.dart';

@JsonSerializable()
class Section {
  int id;
  String name;
  String userId;
  int placeId;
  List<Device> devices;

  Section({
    required this.id,
    required this.name,
    required this.userId,
    required this.placeId,
    required this.devices,
  });

  factory Section.fromJson(Map<String, dynamic> json) =>
      _$SectionFromJson(json);

  Map<String, dynamic> toJson() => _$SectionToJson(this);
}
